UnitTests/Regressions/IssueReportsFixture.cs

```diff
@@ -1631,6 +1631,63 @@ public class GenericClass2<T> : BaseType<T>
 
 		#endregion
 
+		#region 467
+
+		public class Issue467
+		{
+			public interface ITest
+			{
+				int Method();
+			}
+
+			[Fact]
+			public async Task Test()
+			{
+				const int length = 20;
+
+				var m = new Mock<ITest>();
+
+				// Setups the mock to return a sequence of results.
+				var seqSetup = m.SetupSequence(x => x.Method());
+				for (int i = 1; i <= length; i++)
+				{
+					seqSetup = seqSetup
+						.Returns(i);
+				}
+
+				// Queue to store the invocations of the mock.
+				var invocationsQueue = new System.Collections.Concurrent.ConcurrentQueue<int>();
+
+				// We will invoke the mock from two different tasks.
+				var t1 = Task.Run(() => RegisterInvocations(m, invocationsQueue));
+				var t2 = Task.Run(() => RegisterInvocations(m, invocationsQueue));
+				await Task.WhenAll(t1, t2);
+
+				var orderedInvocations = invocationsQueue.OrderBy(x => x).ToArray();
+
+				// The assertion prints the real invocations, you will see duplicates there!
+				Assert.Equal(
```

Comment:
@icalvo's original repro code used MSTest's `CollectionAssert.AreEqual`. Make sure that this is indeed xUnit.NET's way of comparing two sequences.